js/app/modules/selection/articlesForSetCrossSection.js

```diff
@@ -16,24 +16,18 @@ const ArticlesForSetCrossSection = new Module.Class({
     _init: function (props={}) {
         this.parent(props);
         this._set_needs_refresh(true);
-    },
-
-    // Selection.Xapian implementation
-    on_history_changed: function (history) {
-        let item = history.get_current_item();
-        if (item.model instanceof SetObjectModel.SetObjectModel &&
-            item.model !== this._current_set) {
-            this._current_set = item.model;
```

Comment:
@tchx84 the lines of code here are what were actually causing the breakage with the travel app filtering.

The problem was that the old `articlesForSetCrossSection` would only have the correct state after a history change, where it would pick up on the new `_current_set` and track it. But in the case where these selections are created dynamically (which happens in the travel app), there could be a "current set" in our history that predates the creation of this selection.

So basically
 - history changes to reflect current set--Asia
 - `articlesForSetCrossSection` is created, `this._current_set` is null, `this.model` is "Beautiful Beaches" 
 - query is run with null `_current_set`, so we get all beautiful beaches not just "beautiful beaches in asia"


 The fix for this is a bit subtle, but the one I liked the most. We remove any attempt to duplicate and track state from the history store in this module, and just write a getter directly on the history store that retrieves the state we need. That was no matter when this selection is created, `HistoryStore.get_default().current_set` will be the correct set.

This was a change I'd been eyeing for a while, so just did a big refactor of selections while doing that, but other selection changes are not specifically needed for this bug. This clears up a number of theoretical bugs that don't really show up in practice with out yaml files, such as a `search` selection being dynamically created after a search query was entered and not knowing the correct query to use.